src/internals/legend.js

```diff
@@ -513,15 +515,47 @@ extend(ChartInternal.prototype, {
 			.attr("x", $$.isLegendRight || $$.isLegendInset ? xForLegendRect : -200)
 			.attr("y", $$.isLegendRight || $$.isLegendInset ? -200 : yForLegendRect);
 
-		l.append("line")
-			.attr("class", CLASS.legendItemTile)
-			.style("stroke", $$.color)
-			.style("pointer-events", "none")
-			.attr("x1", $$.isLegendRight || $$.isLegendInset ? x1ForLegendTile : -200)
-			.attr("y1", $$.isLegendRight || $$.isLegendInset ? -200 : yForLegendTile)
-			.attr("x2", $$.isLegendRight || $$.isLegendInset ? x2ForLegendTile : -200)
-			.attr("y2", $$.isLegendRight || $$.isLegendInset ? -200 : yForLegendTile)
-			.attr("stroke-width", config.legend_item_tile_height);
+		const hasCustomPoints = $$.config.point_pattern.length;
+
+		if (!hasCustomPoints) {
```

Comment:
Isn't a big deal, but to be more clear the condition block can be written in opposite way.

```js
if (hasCustomPoints) {
    const ids = [];
    l.append(d => {
        ...
} else {
    l.append("line")
        ...
}
```